nearcore/src/config.rs

```diff
@@ -487,10 +487,24 @@ impl Default for Config {
 
 impl Config {
     pub fn from_file(path: &Path) -> anyhow::Result<Self> {
+        let mut unrecognised_fields = Vec::new();
         let s = std::fs::read_to_string(path)
             .with_context(|| format!("Failed to read config from {}", path.display()))?;
-        let config = serde_json::from_str(&s)
+        let config =
+            serde_ignored::deserialize(&mut serde_json::Deserializer::from_str(&s), |path| {
+                unrecognised_fields.push(path.to_string());
+            })
             .with_context(|| format!("Failed to deserialize config from {}", path.display()))?;
+
+        if unrecognised_fields.len() > 0 {
+            warn!(
+                "While trying to deserialize a config from {},
+		the following unrecognised fields were encountered: {:?}",
```

Comment:
This should be single line. You can terminate line with ` \` if you want to wrap it. But then also the alignment is kinda weird.